src/ape/api/networks.py

```diff
@@ -576,20 +577,36 @@ class ProviderContextManager(ManagerAccessMixin):
     connected_providers: Dict[str, "ProviderAPI"] = {}
     provider_stack: List[str] = []
     disconnect_map: Dict[str, bool] = {}
+    recycled_provider: ClassVar[Optional["ProviderAPI"]] = None
```

Comment:
This needs comments everywhere

Reply:
added comments explaining the goal and use